src/components/ChatEntry.js

```diff
@@ -1,22 +1,43 @@
 import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp'
 
 const ChatEntry = (props) => {
+
+
+  const updateLikeButtonClick = () =>{
+    const updateChat ={
+      id : props.id,
+      sender : props.sender,
+      body : props.body,
+      liked : !props.liked, 
+      timeStamp : props.timeStamp,
+    }
+    props.onUpdate(updateChat);
+  }
+
+  const heartColor = props.liked ? '❤️': '🤍'; 
+
   return (
     <div className="chat-entry local">
-      <h2 className="entry-name">Replace with name of sender</h2>
+      <h2 className="entry-name">{props.sender}</h2>
       <section className="entry-bubble">
-        <p>Replace with body of ChatEntry</p>
-        <p className="entry-time">Replace with TimeStamp component</p>
-        <button className="like">🤍</button>
+        <p> {props.body} </p>
+        <p className="entry-time"><TimeStamp time = {props.timeStamp}></TimeStamp></p>
+        <button onClick ={updateLikeButtonClick} className="like">{heartColor}</button>
       </section>
     </div>
   );
 };
 
 ChatEntry.propTypes = {
-  //Fill with correct proptypes
+  id: PropTypes.number.isRequired,
+  sender: PropTypes.string.isRequired,
+  body: PropTypes.string.isRequired,
+  liked: PropTypes.bool.isRequired,
+  timeStamp: PropTypes.string.isRequired,
+  onUpdate: PropTypes.func.isRequired
```

Comment:
Because this proptype is marked `isRequired`, we're getting warnings in the tests, since the test author didn't know what you were going to call this property, and so it's absent from the props passed in during the tests. To avoid the warnings, we can leave off `isRequired`.

Ideally, we'd also include code to ensure that any "optional" values are actually present before trying to use them. For example, in the like click handler we could do something like
```js
    if (onUpdate) {
        onUpdate(id);
    }
```
which would only try to call the function if it's not falsy (`undefined` is falsy).